girder/models/model_base.py

```diff
@@ -45,6 +45,44 @@
 _modelSingletons = []
 
 
+def _permissionClauses(user=None, level=None, prefix=''):
+    """
+    Given a user and access level, return a list of clauses that can be used as
+    part of a Mongo find query or aggregate match step.
+
+    :param user: The user to check policies against.
+    :type user: dict or None
+    :param level: The access level.  Explicitly passing None skips doing
+        permissions checks.
+    :type level: AccessType
+    :param prefix: an optional string to prepend to the keys used in the
+        clauses.
+    :type prefix: str
+    :returns: A query dictionary with an '$or' entry which consists of a list
+        of match clauses, any one of which implies validation.
+    """
+    permissionClauses = []
+    if level is None or (user and user['admin']):
+        # Without a level or with an admin user, match everything.
+        permissionClauses.append({})
```

Comment:
Don't we want to `return` here? If `level=None` but we have a user, the below block will execute with `'$gte': None`, which seems like it would cause an error.

Reply:
I'll change this to return here, too.